### DISCUSSION TITLE
localhost/IP地址外部直接链接如何开启https

### DISCUSSION THREAD

**Discussion by sacrrie** [xcatliu/chatgpt-next#70] [Q&A] [answered]
不想设置域名，docker部署在自己的服务器上的，如何开启https呢？

**Comment by xcatliu** [comment, marked as answer]
我是通过 https://letsencrypt.org/zh-cn/getting-started/ 自动配置好 https 的。可以参考[这个文档](https://letsencrypt.org/zh-cn/getting-started/) 安装 Certbot 然后执行一个脚本就搞定了

ip 地址应该也支持配置 https，不过我不确定免费的 [Let’s Encrypt](https://letsencrypt.org/zh-cn/getting-started/) 是否支持，你可以试一下。

**Reply by sacrrie** [reply]
IP 不好办，我去注册了个域名，按照各种说明配置好nginx后，edge浏览器仍然说不安全，因为网站的部分仍然不安全。这是为啥啊... 一样的docker你的域名为啥一切正常。而且我在手机上打开网站是nginx说要继续配置。我太小白了，能再详细介绍一下你怎么设置吗？

**Reply by xcatliu** [reply]
@sacrrie 网站发来看看

**Reply by sacrrie** [reply]
解决了，参照你的nginx配置文档修改了我的，然后重启了下nginx好了..... 前端果然很深奥... 不好意思添麻烦了
